service/worker/workerdeployment/version_workflow.go

```diff
@@ -209,7 +210,7 @@ func (d *VersionWorkflowRunner) run(ctx workflow.Context) error {
 	d.logger.Debug("Version doing continue-as-new")
 	nextArgs := d.WorkerDeploymentVersionWorkflowArgs
 	nextArgs.VersionState = d.VersionState
-	return workflow.NewContinueAsNewError(ctx, VersionWorkflow, nextArgs)
```

Comment:
did `VersionWorkflow` not work? like, is passing `WorkerDeploymentVersionWorkflowType` an optimization or a bug fix (just curious)

Reply:
I think in simple situations it works, but once you start overriding the name in register call (which we do) and especially if you start wrapping the wf func in a closure etc it becomes dangerous IMO.